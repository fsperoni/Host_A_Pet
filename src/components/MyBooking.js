import React, { useContext, useEffect, useState } from 'react';
import UserContext from '../hooks/useUserContext';
import ReviewForm from './ReviewForm';

const MyBooking = ({ booking, reviews }) => {
  const [showReviewForm, setShowReviewForm] = useState(false);
  const [review, setReview] = useState([]);
  const { currentUser } = useContext(UserContext);

  useEffect(function getReview() {
    const bookingReview = reviews.filter(review => (
      review.reviewer === currentUser.username &&
      review.reviewee === booking.user.username))
    setReview(bookingReview);
  }, [reviews]);

  const handleReview = () => {
    setShowReviewForm(!showReviewForm);
  }
  return (
    <div className="card mb-3">
      <div className="card-header">
        {booking.role1}
      </div>
      <div className="card-body">
        <h5 className="card-title">Booking Details:</h5>
        <p>Dates: {booking.startDate} - {booking.endDate}</p>
        <p>{booking.role2}: {booking.user.username}</p>
        <p>User Ratings: {booking.user.rating}</p>
        <button className="btn btn-primary" onClick={handleReview}>
          Review
        </button>
        {showReviewForm ?
          <ReviewForm review={review[0]} setReview={setReview} 
          setShowReviewForm={setShowReviewForm} reviewee={booking.user.username} /> :
          null
        }
      </div>
    </div>
  )
}

export default MyBooking;