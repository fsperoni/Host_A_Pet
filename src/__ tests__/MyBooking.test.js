import { render } from "@testing-library/react";
import MyBooking from "../components/MyBooking";
import UserContext from "../hooks/useUserContext";

const {
  currentAdminUser,
  reviews, booking
} = require("./_testCommon");

// smoke test
it('renders without crashing', function() {
  render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <MyBooking reviews={reviews()} booking={booking()} 
        />
    </UserContext.Provider>);
});

//snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <MyBooking reviews={reviews()} booking={booking()} 
        />
    </UserContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Booking Details:' in the component.", function() {
  const { getByText } = render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <MyBooking reviews={reviews()} booking={booking()} 
        />
    </UserContext.Provider>
  );
  expect(getByText("Booking Details:")).toBeInTheDocument();
});
