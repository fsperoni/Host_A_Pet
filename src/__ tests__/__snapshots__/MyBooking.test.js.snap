// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card mb-3"
  >
    <div
      class="card-header"
    >
      Host
    </div>
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Booking Details:
      </h5>
      <p>
        Dates: 2022-04-16 - 2022-04-30
      </p>
      <p>
        Pet Owner: testadmin
      </p>
      <p>
        User Ratings: 4.50
      </p>
      <button
        class="btn btn-primary"
      >
        Review
      </button>
    </div>
  </div>
</DocumentFragment>
`;
